File: backend/src/main/kotlin/ru/itmo/db/coursework/api/mapping/TeaInstanceDtoMapper.kt
package ru.itmo.db.coursework.api.mapping

import org.springframework.stereotype.Service
import ru.itmo.db.coursework.api.dto.TeaInstanceDto
import ru.itmo.db.coursework.model.TeaInstance

@Service
class TeaInstanceDtoMapper {
    fun fromDto(dto: TeaInstanceDto) = with(dto, {
        TeaInstance(
            teaId = teaId,
            amount = amount
        )
    })

    fun toDto(model: TeaInstance) = with(model, {
        TeaInstanceDto(
            teaId = teaId,
            amount = amount
        )
    })
}

File: frontend/src/api/mapping/TeaInstanceDtoMapper.ts
import TeaDto from "$src/api/dto/TeaDto";
import Tea from "$src/game/model/Tea";
import CircuitBoardInstanceDto from "$src/api/dto/CircuitBoardInstanceDto";
import CircuitBoardInstance from "$src/game/model/CircuitBoardInstance";
import TeaInstanceDto from "$src/api/dto/TeaInstanceDto";
import TeaInstance from "$src/game/model/TeaInstance";

export default class TeaInstanceDtoMapper {
  public static fromDto(dto: TeaInstanceDto) {
    return new TeaInstance(dto.tea_id, dto.amount)
  }

  public static toDto(model: TeaInstance) {
    return new TeaInstanceDto(model.teaId, model.amount)
  }
}